src/ingest_validation_tests/publication_validator.py

```diff
@@ -0,0 +1,56 @@
+"""
+Test for some common errors in the directory and file structure of publications.
+"""
+
+from typing import List
+import frontmatter
+from ingest_validation_tools.plugin_validator import Validator
+
+
+class PublicationValidator(Validator):
+    """
+    Test for some common errors in the directory and file structure of
+    publications.
+    """
+    description = "Test for common problems found in publications"
+    cost = 1.0
+    def collect_errors(self, **kwargs) -> List[str]:
+        """
+        Return the errors found by this validator
+        """
+        if self.assay_type != 'Publication':
+            return []  # We only test Publication data
+        rslt = []
+        try:
+            vignette_path = self.path / 'vignettes'
+            assert vignette_path.is_dir(), 'vignettes not found or not a directory'
+            for this_vignette_path in vignette_path.glob('*'):
+                assert this_vignette_path.is_dir(), (f"Found the non-dir {this_vignette_path}"
+                                                     " in vignettes")
+                this_vignette_all_paths = set(this_vignette_path.glob('*'))
+                if not all(pth.is_file() for pth in this_vignette_all_paths):
+                    raise AssertionError('Found a subdirectory in a vignette')
+                md_found = False
+                vig_figures = []
+                for md_path in this_vignette_path.glob('*.md'):
+                    if md_found:
+                        raise AssertionError('A vignette has more than one markdown file')
+                    else:
+                        md_found = True
+                    vig_fm = frontmatter.loads(md_path.read_text())
+                    for key in ['name', 'figures']:
```

Comment:
This might be too fiddly but a problem here is that if the .md file is not correctly formatted, you get the "vignette markdown has no {key}" message when the real issue is that the spacing is off. I don't see an obvious way to detect this using frontmatter (I may well be wrong though). Maybe just adding to the exception message would help with diagnosis? e.g. 'vignette markdown has no {key}; if {key} key exists in .md file, check that formatting is correct'

Reply:
I've changed the error message to hopefully address this.